src/core/views.py

```diff
@@ -2629,3 +2635,115 @@ def __init__(self, *args, **kwargs):
+class BaseUserList(GenericFacetedListView):
+
+    model = core_models.Account
+    template_name = 'core/manager/users/list.html'
+
+    def get_facets(self):
+        facets = {
+            'q': {
+                'type': 'search',
+                'field_label': 'Search',
+            },
+            'is_active': {
+                'type': 'boolean',
+                'field_label': 'Active',
+            },
+            'is_staff': {
+                'type': 'boolean',
+                'field_label': 'Staff member',
+            },
+            'accountrole__role__pk': {
+                'type': 'foreign_key',
+                'model': models.Role,
+                'field_label': 'Role',
+                'choice_label_field': 'name',
+            },
+            'accountrole__journal__pk': {
+                'type': 'foreign_key',
+                'model': journal_models.Journal,
+                'field_label': 'Journal',
+                'choice_label_field': 'name',
+            },
+        }
+        return self.filter_facets_if_journal(facets)
+
+    def get_order_by_choices(self):
+        return [
+            ('-date_joined', _('Newest')),
+            ('date_joined', _('Oldest')),
+            ('last_name', _('Last name A-Z')),
+            ('-last_name', _('Last name Z-A')),
+        ]
+
+    def get_journal_filter_query(self):
+        if self.request.journal:
+            return Q(accountrole__journal=self.request.journal)
+        else:
+            return Q()
+
+    def filter_facets_if_journal(self, facets):
+        if self.request.journal:
+            facets.pop('accountrole__journal__pk', '')
+            facets.pop('is_staff', '')
+            return facets
+        else:
+            return facets
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        roles = models.Role.objects.exclude(slug='reader')
+        if not self.request.user.is_staff:
+            roles = roles.exclude(slug='journal-manager')
+        context['roles'] = roles
+
+        accountrole_form = forms.AccountRoleForm({
+            'journal': self.request.journal,
+        })
+        if self.request.journal:
+            accountrole_form.fields['journal'].widget.choices = [
+                (self.request.journal.pk, self.request.journal.name)
+            ]
+        else:
+            journal_names = core_models.SettingValue.objects.filter(
+                setting__group__name='general',
+                setting__name='journal_name',
+                journal__isnull=False,
+            ).order_by('value')
+            choices = [(None, '---------')]
+            choices.extend([(n.journal.pk, n.value) for n in journal_names])
+            accountrole_form.fields['journal'].widget.choices = choices
+        context['accountrole_form'] = accountrole_form
+
+        return context
+
+    def post(self, request, *args, **kwargs):
+
+        post_data = request.POST.copy()
```

Comment:
Rather than copy post data so that journal can be inserted into it would be good to handle this in the same way we have before by passing `journal` into the form as a kwarg and handle it in the `__init__` and `save` methods.

Reply:
Thanks @ajrbyers -- turns out those lines were only needed by an earlier draft of this, so I've cleaned them up.